### PR TITLE
Preserve Float type for Periodic Kernel

### PR DESCRIPTION
**Pull request by molet**
**Summary**
Tiny change to preserve float type of inputs and parameters. For most of the kernels, if the inputs and parameters have Float32 type, then the output has also Float32:
```julia
julia> v1 = 0.9f0; v2 = 0.7f0;
julia> SqExponentialKernel()(v1, v2)
0.9801987f0
julia> GammaRationalKernel(alpha=2.0f0, gamma=1.0f0)(v1, v2)
0.82644624f0
```
However, currently, `PeriodicKernel` doesn't support this:
```julia
julia> PeriodicKernel(r=[1.0f0])(v1, v2)
0.8413515018155655
```
The simple change of `exp(-0.5d)` to `exp(-d / 2)` in its `kappa` function fixes it:
```julia
julia> PeriodicKernel(r=[1.0f0])(v1, v2)
0.8413515f0
```

### CODE REVIEW

**Review by devmotion — COMMENTED** [review]
Looks good to me, can you add a test for the fix?

**Comment by molet** [comment]
Certainly, it's been added.

File: test/basekernels/periodic.jl [hunk 1]
```diff
@@ -8,6 +8,7 @@
     @test k(v1, v2) ≈ exp(-0.5 * sum(abs2, sinpi.(v1 - v2) ./ r))
     @test k(v1, v2) == k(v2, v1)
     @test PeriodicKernel(3)(v1, v2) == PeriodicKernel(; r=ones(3))(v1, v2)
+    @test isa(PeriodicKernel(r=Float32.(r))(Float32.(v1), Float32.(v2)), Float32)
```
Comment (devmotion) [inline test/basekernels/periodic.jl:11]: Based on the logs JuliaFormatter would like
```suggestion
    @test isa(PeriodicKernel(; r=Float32.(r))(Float32.(v1), Float32.(v2)), Float32)
```
Alternatively, possibly a bit easier to read:
```suggestion
    @test PeriodicKernel(; r=Float32.(r))(Float32.(v1), Float32.(v2)) isa Float32
```

**Review by willtebbutt — APPROVED** [review]
LGTM

**Comment by willtebbutt** [comment]
The changes that you've made do not appear to have anything to do with the test failures, so I'm going to merge.

**Pull request merged**